app/services/imports/case_logs_field_import_service.rb

```diff
@@ -4,16 +4,38 @@ def update_field(field, folder)
       case field
       when "tenant_code"
         import_from(folder, :update_tenant_code)
+      when "major_repairs"
+        import_from(folder, :update_major_repairs)
       else
         raise "Updating #{field} is not supported by the field import service"
       end
     end
 
   private
 
+    def update_major_repairs(xml_doc)
+      record = log(xml_doc)
+
+      if record.present?
+        previous_status = field_value(xml_doc, "meta", "status")
+        major_repairs_date = compose_date(xml_doc, "MRCDAY", "MRCMONTH", "MRCYEAR")
+        major_repairs = if major_repairs_date.present? && previous_status.include?("submitted")
+                          1
+                        elsif previous_status.include?("submitted")
+                          0
+                        end
+        if major_repairs.present? && record.majorrepairs.blank? && record.status != "completed"
```

Comment:
There are a few cases where this block would skip with no logs.
* Partially submitted logs (saved status)
* Completed logs on our side with no `majorrepairs` value